### PR TITLE
feat(boards): #MAG-7 add board properties popup

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- add board properties popup to update a board
- add properties button to toaster

https://confluence.support-ent.fr/display/RN/Boards

## Checklist tests

- check if updating boards is behaving correctly

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-7

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by alicedraillard — APPROVED** [review]

File: src/main/java/fr/cgi/magneto/model/BoardUpdate.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package fr.cgi.magneto.model;
+
+import fr.cgi.magneto.core.constants.*;
+import fr.cgi.magneto.helper.*;
+import io.vertx.core.json.*;
+
+import java.util.*;
+
+public class BoardUpdate implements Model<BoardUpdate> {
```
Comment (JinDouang) [inline src/main/java/fr/cgi/magneto/model/BoardUpdate.java:9]: A renommer le model (pas seulement update)

File: src/main/java/fr/cgi/magneto/model/BoardUpdate.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getModificationDate() {
+        return modificationDate;
+    }
+
+    public void setModificationDate(String modificationDate) {
+        this.modificationDate = modificationDate;
```
Comment (JinDouang) [inline src/main/java/fr/cgi/magneto/model/BoardUpdate.java:51]: tu pourrais retourner l'objet

File: src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java [hunk 3]
```diff
@@ -47,6 +47,25 @@ public Future<JsonObject> create(UserInfos user, JsonObject board) {
...
+                .put(Field.OWNERID, user.getUserId());
+        JsonObject update = new JsonObject().put("$set", board.toJson());
+        mongoDb.update(this.collection, query, update, MongoDbResult.validActionResultHandler(results -> {
+            if (results.isLeft()) {
+                String message = String.format("[Magneto@%s::update] Failed to update board", this.getClass().getSimpleName());
+                log.error(String.format("%s : %s", message, results.left().getValue()));
+                promise.fail(message);
+                return;
+            }
+            promise.complete();
```
Comment (JinDouang) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java:64]: return Void si tu comptes rien compléter

File: src/main/resources/public/ts/directives/board-list/board-list.directive.ts [hunk 4]
```diff
@@ -32,11 +34,19 @@ class Controller implements IViewModel {
...
     selectBoard = (boardId: string): void => {
+        // select board
         if (!this.isBoardSelected(boardId)) {
             this.selectedBoardIds.push(boardId);
         } else {
+            // deselect board
             this.selectedBoardIds = this.selectedBoardIds.filter((id: string) => id !== boardId);
         }
+
+        let selectedUpdateBoard: Board = this.boards.find(board => board.id === this.selectedBoardIds[0]);
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/board-list/board-list.directive.ts:45]: Typage board dans ton .find

File: src/main/resources/public/ts/models/board.model.ts [hunk 5]
```diff
@@ -25,11 +25,81 @@ export interface IBoardsParamsRequest {
...
+
+    isValid(): boolean {
+        return this.title !== null && this.title !== '' &&
+            this.description != null && this.description !== '' &&
+            this.imageUrl != null && this.imageUrl !== ''
+    }
+
+    toJSON(): IBoardPayload {
+        if (!this.id || this.id == '') {
+            return {
```
Comment (JinDouang) [inline src/main/resources/public/ts/models/board.model.ts:87]: Tu peux le refactoriser avec un let json et tu le return

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java [hunk 6]
```diff
@@ -47,6 +47,25 @@ public Future<JsonObject> create(UserInfos user, JsonObject board) {
         return promise.future();
     }
 
+    @Override
+    public Future<JsonObject> update(String boardId, UserInfos user, BoardUpdate board) {
+        Promise<JsonObject> promise = Promise.promise();
+        JsonObject query = new JsonObject()
+                .put(Field._ID, boardId)
+                .put(Field.OWNERID, user.getUserId());
+        JsonObject update = new JsonObject().put("$set", board.toJson());
```
Comment (BastienBenard) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultBoardService.java:56]: Est ce que ce serait pas mieux de faire des constantes pour les clé mongo ? ($set, $size etc...)

File: src/main/resources/public/ts/services/__test__/boards.service.test.ts [hunk 7]
```diff
@@ -39,20 +39,21 @@ describe('BoardsService', () => {
         const data = {
             response: true
         }
-        const params = {
-            title: 'title',
-            description: 'description',
-            imageUrl: 'imageUrl'
-        }
+
+        let form = new BoardForm();
```
Comment (BastienBenard) [inline src/main/resources/public/ts/services/__test__/boards.service.test.ts:43]: typage de form

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**